lib/core/src/irods_client_negotiation.cpp

```diff
@@ -60,7 +60,7 @@ namespace irods
 
                 const auto hostname_in_provider_hosts_list = std::any_of(
                     std::cbegin(*provider_hosts), std::cend(*provider_hosts), [&_host_name](const auto& _host) {
-                        return _host_name == _host;
+                        return _host_name == static_cast<std::string>(_host);
```

Comment:
Should I call `_host.get_ref<std::string>()` instead?

Reply:
Use `_host.get_ref<const std::string&>()`.

Here's an example of the JSON structure. Taken from https://docs.irods.org/4.3.4/system_overview/federation/#zoneas-server_configjson.
```json
"federation": [
    {
        "catalog_provider_hosts": ["zoneB-provider.example.org"], 
        "negotiation_key": "_____32_byte_pre_shared_key_____",
        "zone_key": "ZONE_KEY_FOR_ALL_OF_ZONE_B", 
        "zone_name": "ZoneB", 
        "zone_port": 2247
    }
]
```

Reply:
This worked in the development environment, but does not work in CI:
```
/__w/irods/irods/lib/core/src/irods_client_negotiation.cpp:63:52: error: use 'template' keyword to treat 'get_ref' as a dependent template name
                        return _host_name == _host.get_ref<const std::string&>();
                                                   ^
                                                   template 
```

That's concerning.